_posts/2021-10-20-cocotb-1-6-0.md

```diff
@@ -0,0 +1,184 @@
+Only then can all other coroutines, e.g. `coro3()` do their work.
+
+We deprecated `cocotb.fork()` because it has slightly odd behavior (by today's standards, of course):
+* `cocotb.fork()` gives up control, but isn't a async function which needs an `await` in front of it.
+* `cocotb.fork()` yields control "only a bit" to let exactly one other coroutine execute, but in a way which is rather unique.
+* And finally there are a couple other corner cases in error handling which are very hard or impossible to get right with `cocotb.fork()`, and which many of our users have painfully experienced over the years.
+
+**`cocotb.fork()` is deprecated, but it will remain available in cocotb 1.x.
+We encourage you to migrate your testbenches to the new scheduling routines, most likely by simply replacing `cocotb.fork()` with `cocotb.start_soon()`.
+Have a look at [PR #2713](https://github.com/cocotb/cocotb/pull/2713/files) for how we did it the conversion in cocotb itself.**
+
+
+### Further reading
+
+If you'd like to know more about coroutines and scheduling in cocotb have a look at the [Coroutines and Tasks section](https://docs.cocotb.org/en/v1.6.0/coroutines.html#coroutines) in the documentation (which also explains the `cocotb.create_task()` function, which we skipped here).
+
+## They are your type: HDL datatypes
+
+Datatypes are behind every line of code in your editor.
+When you write `logic [3:0] my_signal` in SystemVerilog datatypes are clearly visible.
+But datatypes are there even if you don't explicitly write them down, as in `my_signal <= -2` in cocotb.
+
+Right now, cocotb is [fairly relaxed about HDL datatypes](https://docs.cocotb.org/en/v1.6.0/writing_testbenches.html#reading-values-from-signals), and freely converts between them as needed mostly through its `BinaryValue` class (which is behind the `my_signal` assignment seen before).
+However, being relaxed isn't always helpful, as some cocotb users have found out: conversions can go wrong, and expressiveness can be be lost.
+
+Starting with cocotb 1.6, we're introducing datatypes which model common HDL data types: `Logic` for four-state logic values, `Bit` for two-state logic, `Array` and `LogicArray` for HDL-style arrays or vectors.
+All these new types have been carefully designed to feel natural for both VHDL and SystemVerilog programmers, and to provide type safety and useful typecasts.
+
+As of now these datatypes [can be used in custom testbench code](https://docs.cocotb.org/en/v1.6.0/library_reference.html#hdl-datatypes).
+Going forward, we plan to bake these types more deeply into cocotb, and make signal handles return them.
+
+## Assigning values to signals using `<=` is deprecated
+
+One of the unique, but also slightly obscure, features of cocotb is on its way out:
+Setting values on handles using the `your_signal <= newval` syntax is now deprecated.
+Please use `your_signal.value = newval` instead.
```

Comment:
```suggestion
Setting values on handles using the `my_signal <= newval` syntax is now deprecated.
Please use `my_signal.value = newval` instead.
```
We use `my_signal`already a bit further up... unless this is meant jokingly?

Reply:
no pun intended, good catch!